src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,378 @@
+  2[x∸y] + pred (suc 1ᵇ + 2y)           ≡⟨ cong ((2[x∸y] +_) ∘ pred) (suc-+ 1ᵇ 2y) ⟩
+  2[x∸y] + pred (suc (1ᵇ + 2y))         ≡⟨ cong (2[x∸y] +_) (pred-suc (1ᵇ + 2y)) ⟩
+  2[x∸y] + (1ᵇ + 2y)                    ≡⟨ x∙yz≈y∙xz 2[x∸y] 1ᵇ 2y ⟩
+  1ᵇ + (2[x∸y] + 2y)                    ≡⟨ cong (1ᵇ +_) $ sym $ double-distrib-+ x∸y y ⟩
+  1ᵇ + double (x∸y + y)                 ≡⟨ cong ((1ᵇ +_) ∘ double) $ sym $
+                                           x≥y⇒x≡[x∸y]+y (inj₁ y<x) ⟩
+  1ᵇ + 2x                               ≡⟨ sym (suc≗1+ 2x) ⟩
+  suc 2x                                ≡⟨ sym (1+[2_]-suc-double x) ⟩
+  x'                                    ∎
+  where
+  open ≤-Reasoning
+  x∸y = x ∸ y;  2x = double x;  2y = double y;  2[x∸y] = double x∸y;  x∸y>0 = x<y⇒0<y∸x y<x
+  x∸y≢0       = >⇒≢ x∸y>0
+  2[x∸y]≢0    = x≢0⇒double[x]≢0 x∸y≢0
+  2[suc[y]]≢0 = x≢0⇒double[x]≢0 (suc≢0 {y})
+
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {y'@(1+[2 y ])} (inj₁ (odd<odd y<x)) =  sym $ begin-equality
+  (x' ∸ y') + y'                 ≡⟨ cong ((x' ∸ y') +_) (1+[2_]-suc-double y) ⟩
+  double x∸y + (suc (double y))  ≡⟨ x+suc[y]≡suc[x]+y (double x∸y) (double y) ⟩
+  suc (double x∸y) + double y    ≡⟨ suc-+ (double x∸y) (double y) ⟩
+  suc (double x∸y + double y)    ≡⟨ cong suc $ sym $ double-distrib-+ x∸y y ⟩
+  suc (double (x∸y + y))         ≡⟨ sym (1+[2_]-suc-double _) ⟩
+  1+[2 (x∸y + y) ]               ≡⟨ cong 1+[2_] $ sym $ x≥y⇒x≡[x∸y]+y (inj₁ y<x) ⟩
+  1+[2 x ]                       ∎
+  where open ≤-Reasoning;  x∸y = x ∸ y
+
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {1+[2 .x ]} (inj₂ refl) =  sym $ begin-equality
+  (x' ∸ x') + x'  ≡⟨ cong (_+ x') (x∸x≡0 x') ⟩
+  0ᵇ + x'         ≡⟨ sym (+-identityˡ x') ⟩
+  x'              ∎
+  where open ≤-Reasoning
+
+[x+y]∸y≡x :  ∀ x y → (x + y) ∸ y ≡ x
+[x+y]∸y≡x x y =  sym x≡d
+  where
+  d       = (x + y) ∸ y;           y≤x+y = x≤y+x y x
+  x+y≡d+y = x≥y⇒x≡[x∸y]+y y≤x+y;   x≡d   = +-cancelʳ-≡ x d x+y≡d+y
+
+[x+y]∸x≡y :  ∀ x y → (x + y) ∸ x ≡ y
+[x+y]∸x≡y x y =  trans (cong (_∸ x) (+-comm x y)) ([x+y]∸y≡x y x)
+
+∸-identityʳ :  RightIdentity 0ᵇ _∸_
+∸-identityʳ x =  trans (cong (_∸ 0ᵇ) $ sym $ +-identityʳ x) ([x+y]∸y≡x x 0ᵇ)
+
+x+[y∸x]≡y :  ∀ {x y} → x ≤ y → x + (y ∸ x) ≡ y
+x+[y∸x]≡y {x} {y} x≤y =  begin-equality
+  x + (y ∸ x)   ≡⟨ +-comm x _ ⟩
+  (y ∸ x) + x   ≡⟨ sym (x≥y⇒x≡[x∸y]+y x≤y) ⟩
+  y             ∎
+  where open ≤-Reasoning
+
+[x∸y]+y≡x :  ∀ {x y} → y ≤ x → (x ∸ y) + y ≡ x
+[x∸y]+y≡x {x} {y} y≤x =  trans (+-comm (x ∸ y) y) (x+[y∸x]≡y y≤x)
+
+toℕ-homo-∸ :  ∀ x y → toℕ (x ∸ y) ≡ (toℕ x) ℕ.∸ (toℕ y)
+toℕ-homo-∸ x y  with <-cmp x y
+... | tri< x<y _ _ =  begin
+  toℕ (x ∸ y)   ≡⟨ cong toℕ (x≤y⇒x∸y≡0 x≤y) ⟩
+  toℕ 0ᵇ        ≡⟨ refl ⟩
+  0             ≡⟨ sym (ℕₚ.m≤n⇒m∸n≡0 m≤n) ⟩
+  m ℕ.∸ n       ∎
+  where
+  open ≡-Reasoning
+  m = toℕ x;  n = toℕ y;  x≤y = <⇒≤ x<y;  m≤n = toℕ-mono-≤ x≤y
+
+... | tri≈ _ x≡y _ =  begin
+  toℕ (x ∸ y)  ≡⟨ cong toℕ (x≤y⇒x∸y≡0 {x} {y} x≤y) ⟩
+  toℕ 0ᵇ       ≡⟨ refl ⟩
```

Comment:
`≡⟨ refl ⟩` can be replaced with `≡⟨⟩` (also applies to other lines in the file).